src/rust/aggregator/src/pipelines/spreads.rs

```diff
@@ -0,0 +1,373 @@
+            )
+            .fetch_all(transaction as &mut PgConnection)
+            .await
+            .map_err(to_pipeline_error)?;
+
+            // Get all change events that happened between the last transaction included in the previous
+            // state and the last transaction that happened before the given timestamp.
+            let changes = sqlx::query_file_as!(
+                Change,
+                "sqlx_queries/spreads/get_change_order_size_events_between_txn_versions.sql",
+                txn_version_of_state,
+                txn_version
+            )
+            .fetch_all(transaction as &mut PgConnection)
+            .await
+            .map_err(to_pipeline_error)?;
+
+            // Get all cancel events that happened between the last transaction included in the previous
+            // state and the last transaction that happened before the given timestamp.
+            let cancels = sqlx::query_file!(
+                "sqlx_queries/spreads/get_cancel_order_events_between_txn_versions.sql",
+                txn_version_of_state,
+                txn_version
+            )
+            .fetch_all(transaction as &mut PgConnection)
+            .await
+            .map_err(to_pipeline_error)?;
+
+            // Insert all places as orders into the state.
+            for place in places {
+                orders.insert((place.market_id.clone(), place.order_id.clone()), place);
+            }
+
+            // Remove all orders that have been cancelled.
+            for cancel in cancels {
+                orders.remove(&(cancel.market_id, cancel.order_id));
+            }
+
+            // Handle fills and changes chronologically.
+            let mut fills_index = 0;
+            let mut changes_index = 0;
+            for _ in 0..(fills.len() + changes.len()) {
+                let fill = fills.get(fills_index);
+                let change = changes.get(changes_index);
+                let (f, c) = match (fill, change) {
+                    (Some(fill), Some(change)) => {
+                        if (fill.txn_version.clone(), fill.event_idx.clone())
+                            < (change.txn_version.clone(), change.event_idx.clone())
+                        {
+                            (Some(fill), None)
+                        } else {
+                            (None, Some(change))
+                        }
+                    }
+                    (None, None) => unreachable!(),
+                    other => other,
+                };
+                match (f, c) {
+                    (Some(fill), None) => {
+                        handle_fill(fill, &mut orders, &mut fills_index);
+                    }
+                    (None, Some(change)) => {
+                        handle_change(change, &mut orders, &mut changes_index);
+                    }
+                    _ => unreachable!(),
+                };
+            }
+
+            txn_version_of_state = txn_version;
+        }
+    }
```

Comment:
No blank line here?

Should probably run `rustfmt`

Reply:
Strangely enough my format on save for VS code doesn't even modify this

Noting that we'll probably want to have a strict Rust formatter/linter at some point to avoid such issues

Bookmarking this comment internally for the CI/CD work we have scoped out